docs/sources/next/using-k6/modules.md

```diff
@@ -152,14 +152,215 @@ Another option to distribute libraries is to use a package manager tool like npm
 
 Although k6 does not resolve node modules, you can utilize a Bundler to load npm dependencies, as shown in the [k6-rollup-example](https://github.com/grafana/k6-rollup-example).
 
-## Using TypeScript
+## Use Node.js modules
+
+{{< admonition type="caution" >}}
+
+k6 isn't Node.js or a browser. Packages that rely on APIs provided by Node.js, for instance the `os` and `fs` modules, won't work in k6. The same goes for browser-specific APIs, such as the `window` object.
+
+{{< /admonition >}}
+
+In a JavaScript project running Node.js, modules are imported using either `import` or `require()`, using the Node.js module resolution algorithm. This means that a user can import modules by name, without providing the full filesystem path to the module. For instance:
+
+```javascript
+import { ClassInAModule } from 'cool-module';
+```
+
+The `import` statement would be automatically resolved by the node resolution algorithm by searching:
+
+- The current directory
+- Any `node_modules` folder in the directory
+- Any `node_modules` folder in a parent directory, up to the closest `package.json` file.
+
+As the implementation of `import` in k6 lacks support for the node module resolution algorithm, Node.js modules that resolve external dependencies need to be transformed into a self-contained, isolated, bundle.
+
+That's done with the help of a bundling tool, such as Webpack, which analyses the test script, identifies all external dependencies, and then creates a self-contained bundle including everything necessary to run the script.
+
+If the test script has no external dependencies, already has them vendored in a k6 compatible way, or only uses ES5.1+ features, using a bundler isn't necessary.
+
+### Pick a bundler
+
+You can use any bundler that supports transpilation. Popular ones include, but are not limited to, [webpack](https://github.com/webpack/webpack), [parcel](https://github.com/parcel-bundler/parcel), [rollup](https://github.com/rollup/rollup) and [browserify](https://github.com/browserify/browserify).
+
```

Comment:
I wonder why https://esbuild.github.io/ wasn't mentioned or chosen - it feels like the most obvious choice for our use case given its speed and golangness.

Maybe @szkiba and @pablochacin will want to maybe do this.

But it is definitely better to have some working instructions, so maybe for another PR later.

Reply:
Since esbuild TypeScript support has been added experimentally to k6, this topic will be worth a rewrite soon. In the case of bundler, I would highlight the use of grafana/k6pack. Simple, golang-based, fast.

Reply:
Anyway, I agree with @mstoykov , this should be another PR...